### PR TITLE
Spike/three js test

### PR DESCRIPTION
**Pull request by karimbeyrouti**

### CODE REVIEW

**Comment by NullVoxPopuli** [comment]
Over here: https://github.com/NullVoxPopuli/ember-contextual-services/pull/3/files#diff-51ff1c7fba91f212e33dff6832eb54c2R79
I've been trying to figure out how to get context implemented.
It's looking like it's not possible with public apis.
However, this works:
```ts
    test('provides context one level down', async function(assert) {
      class FooProvider extends ContextProvider { // aka @ember/component
        provided = 'provided some stuff';
      }
      class Consumer extends Component {
        @context(FooProvider) foo!: FooProvider;
      }
      this.owner.register('component:foo-provider', FooProvider);
      this.owner.register('component:consumer', Consumer);
      this.owner.register('template:components/consumer', hbs`{{this.foo.provided}}`);

      await render(hbs`
        <FooProvider>
          <Consumer />
        </FooProvider>
      `);

      assert.dom(this.element).hasText('provided some stuff');
    });
```

but the moment you put a component in-between, it _can't_ work. 
The apis for linking parent/child components are private and seemingly only used in `@ember/component` in the "curly" component-manager: -- I learned that the managers in https://github.com/emberjs/ember.js/tree/d38232f33381751672165ec1852e7ff902f7e9f5/packages/%40ember/-internals/glimmer/lib/component-managers aren't all public implementations -- the APIs we get access to are only from "Custom Component Manager"

/cc @pzuraq, @gaurav, @miguelcobain

File: app/components/e-threejs-demo/demos/cubes.hbs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<Controls @onUpdate={{this.onCountUpdate}} />
+
+<main>
+  <EThreejs @id={{this.sceneId}} as |s|>
```
Comment (NullVoxPopuli) [inline app/components/e-threejs-demo/demos/cubes.hbs:4]: Thoughts on:
```hbs
<EThree::Scene as |s|>
    <s.Lights.DirectionalLight @position={{this.lightPosition}} />
    <s.Cameras.PerspectiveCamera
      @position={{this.cameraPosition}}
      @aspectRatio={{this.aspectRatio}}
    />

    <s.Object3d @rotation={{this.containerRotation}} as |container|>

      {{#each this.appState.objectProperties as |properties|}}
        <container.Objects.Mesh
          @rotation={{properties.r}}
          @position={{properties.p}}
          @scale={{properties.s}}
          @material={{this.material}}
          @geometry={{this.geometry}}
        />
      {{/each}}

    </s.Object3d>

  </EThree::Scene>
</main>
```
?

**Comment by NullVoxPopuli** [comment]
Just cloned. That lines demo is _fantastic_

**Comment by NullVoxPopuli** [comment]
and the spastic cubes is entertaining too :D

File: app/components/e-threejs-demo/demos/cubes.hbs (hunk 1 above)
Comment (karimbeyrouti) [reply app/components/e-threejs-demo/demos/cubes.hbs:4]: think what bothers me with approach is it feels like two API's: one for the template and another for what yields what. 

Overall for now I would somewhat prefer specifying the `@parent` to yielding complex structures from the scene / objects.

**Review by karimbeyrouti — COMMENTED** [review]
Thank you... Although I mostly copied the lines demo from ThreeJS examples to get something up and running rather quickly... but what surprised me the most is how quick it was to implement that in ember.

**Pull request merged**